fg_lib/include/fg_lib.h

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef __FG_LIB_H__
+#define __FG_LIB_H__
+
+/**
+ * @brief Algorithm for battery state-of-charge estimation ("fuel gauge").
+ *
+ * @details Algorithm requires a model of the specific battery model used in the given application.
+ *          This model, along with measurements of battery voltage, current, and temperature,
+ *          will determine the predicted state-of-charge.
+ *          The state-of-charge, or "SOC", is a percentage value between 0% and 100%.
+ *
+ * @note Current and voltage measurements does not have to be made at a certain interval.
+ *       However, the two values should be measured as closely as possible in time.
+ *
+ * @defgroup fg_lib Battery fuel gauge library
+ * @{
+ */
+
+#include <stddef.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Version number of battery fuel gauge library in format "x.y.z".
+ */
+extern const char *fg_lib_version;
+
+/**
+ * @brief Build date of battery fuel gauge library
+ */
+extern const char *fg_lib_build_date;
+
+/**
+ * @brief Battery model parameters
+ *
+ * @details Parameter details purposefully not documented.
+ */
+struct battery_model {
+	float param_1[201];
+	float temps[3];
+	float param_2[122];
+	float param_3[201];
+	float param_4[201];
+	float param_5[201];
+	float param_6[201];
+	float param_7[122];
+	float param_8[122];
+	float param_9[3];
+	float param_10[6];
+	float param_11[6];
+	float param_12[3];
+	char name[64];
+};
+
+
+/**
+ * @brief Library state information. Useful for debugging.
+ */
+struct fg_lib_state_info {
+	/** Predicted voltage */
+	float yhat;
+	/** Battery resistance */
+	float r0;
+	/** Truncated temperature. Happens when measured temperature falls outside modeled range. */
+	float T_truncated;
+};
+
+/**
+ * @brief Library init parameter struct
+ */
+struct fg_lib_init_parameters {
+	/** Initial voltage [V] */
+	float v0;
+	/** Initial current [A] */
+	float i0;
+	/** Initial temperature [C] */
+	float t0;
+	/** Battery model */
+	const struct battery_model *model;
+};
+
+/**
+ * @brief Initialize the battery fuel gauge library.
+ *
+ * @param[in] parameters Pointer to the parameter structure.
+ * @param[out] v0 Adjusted battery voltage (optional, for logging purposes).
+ *
+ * @retval 0 If successful.
+ * @retval -22 if invalid parameters.
+ */
+int fg_lib_init(const struct fg_lib_init_parameters *parameters, float *v0);
+
+/**
+ * @brief Process battery measurement values.
+ *
+ * @param v Measured battery voltage [V]
+ * @param i Measured battery current [A]
+ * @param T Measured battery temperature [C]
+ * @param t_delta Time delta since previous processed measurement [s]
+ * @param state Pointer to state info struct (optional, set to null to ignore).
+ *
+ * @retval Predicted state-of-charge [%]
```

Comment:
```suggestion
 * @param v Measured battery voltage [V].
 * @param i Measured battery current [A].
 * @param T Measured battery temperature [C].
 * @param t_delta Time delta since previous processed measurement [s].
 * @param state Pointer to state info struct (optional, set to null to ignore).
 *
 * @retval Predicted state-of-charge [%].
```